File: go/leap/leap.go
/*
Package leap provides a method to determine whether a given year is leap year or not.
*/
package leap

// IsLeapYear returns true if it is leap year, otherwise, returns false.
func IsLeapYear(year int) bool {
	return (year%4 == 0) && ((year%400 == 0) || (year%100 != 0))
}

File: fsharp/leap/Leap.fs
module Leap

let isDivisibleBy (year: int) (num: int): bool = year % num = 0

let leapYear (year: int): bool = 
    year |> isDivisibleBy <| 4 && (
        not (year |> isDivisibleBy <| 100) || year |> isDivisibleBy <| 400
    )
